tests/monitoring/vm_monitoring.go

```diff
@@ -0,0 +1,229 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package monitoring
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	virtctlpause "kubevirt.io/kubevirt/pkg/virtctl/pause"
+
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/clientcmd"
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/framework/checks"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/framework/matcher"
+	"kubevirt.io/kubevirt/tests/libnode"
+	"kubevirt.io/kubevirt/tests/libvmi"
+	"kubevirt.io/kubevirt/tests/libwait"
+	"kubevirt.io/kubevirt/tests/testsuite"
+	"kubevirt.io/kubevirt/tests/util"
+)
+
+var _ = Describe("[Serial][sig-monitoring]VM Monitoring", Serial, decorators.SigMonitoring, func() {
+	var err error
+	var virtClient kubecli.KubevirtClient
+
+	BeforeEach(func() {
+		virtClient = kubevirt.Client()
+	})
+
+	Context("VM status metrics", func() {
+		var vm *v1.VirtualMachine
+
+		BeforeEach(func() {
+			vmi := tests.NewRandomVMI()
+			vm = tests.NewRandomVirtualMachine(vmi, false)
+
+			By("Create a VirtualMachine")
+			_, err = virtClient.VirtualMachine(vm.Namespace).Create(context.Background(), vm)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		checkMetricTo := func(metric string, labels map[string]string, matcher types.GomegaMatcher) {
+			EventuallyWithOffset(1, func() int {
+				v, err := getMetricValueWithLabels(virtClient, metric, labels)
+				if err != nil {
+					return -1
+				}
+				i, err := strconv.Atoi(v)
+				Expect(err).ToNot(HaveOccurred())
+				return i
+			}, 3*time.Minute, 20*time.Second).Should(matcher)
```

Comment:
as per https://github.com/kubevirt/kubevirt/pull/9470#discussion_r1146203503
```suggestion
			}, 3*time.Minute, 20*time.Second).Should(matcher, "<add custom description of the failure>")
```

Reply:
added